test/extensions/clusters/aggregate/cluster_test.cc

```diff
@@ -169,6 +169,83 @@ class AggregateClusterTest : public Event::TestUsingSimulatedTime, public testin
 )EOF";
 }; // namespace Aggregate
 
+TEST_F(AggregateClusterTest, CircuitBreakerMaxConnectionsTest) {
+  const std::string yaml_config = R"EOF(
+    name: aggregate_cluster
+    connect_timeout: 0.25s
+    lb_policy: CLUSTER_PROVIDED
+    circuit_breakers:
+      thresholds:
+      - priority: DEFAULT
+        max_connections: 1
+        track_remaining: true
+    cluster_type:
+      name: envoy.clusters.aggregate
+      typed_config:
+        "@type": type.googleapis.com/envoy.extensions.clusters.aggregate.v3.ClusterConfig
+        clusters:
+        - primary
+        - secondary
+)EOF";
+
+  initialize(yaml_config);
+
+  // resource manager for the DEFAULT priority (look above^)
+  Upstream::ResourceManager& resource_manager =
+      cluster_->info()->resourceManager(Upstream::ResourcePriority::Default);
+
+  // get the cx_open stat (this represents the circuit breaker state: closed is 0, open is 1)
+  Stats::StatNameManagedStorage cx_open_stat("circuit_breakers.default.cx_open",
+                                             cluster_->info()->statsScope().symbolTable());
+  Stats::Gauge& cx_open = cluster_->info()->statsScope().gaugeFromStatName(
+      cx_open_stat.statName(), Stats::Gauge::ImportMode::Accumulate);
+
+  // NOTE: in order to use remaining_cx (and for it to not always be 0)
+  // we need to use "track_remaining: true" in the config to get the remaining_cx stat
+  // (this represents the number of remaining connections)
+  Stats::StatNameManagedStorage remaining_cx_stat("circuit_breakers.default.remaining_cx",
+                                                  cluster_->info()->statsScope().symbolTable());
+  Stats::Gauge& remaining_cx = cluster_->info()->statsScope().gaugeFromStatName(
+      remaining_cx_stat.statName(), Stats::Gauge::ImportMode::Accumulate);
```

Comment:
Can't we move this to the `initialize()` function as part of the setup ?

Reply:
will try to give that a go today